tests/Comparator/Int64Comparator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace MongoDB\Tests\Comparator;
+
+use MongoDB\BSON\Int64;
+use SebastianBergmann\Comparator\Comparator;
+use SebastianBergmann\Comparator\ComparisonFailure;
+
+use function is_int;
+use function is_numeric;
+use function is_string;
+use function sprintf;
+
+use const PHP_INT_SIZE;
+
+class Int64Comparator extends Comparator
+{
+    public function accepts($expected, $actual)
+    {
+        // Only compare if either value is an Int64
+        return ($expected instanceof Int64 && $this->isComparable($actual))
+            || ($actual instanceof Int64 && $this->isComparable($expected));
+    }
+
+    public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = false, $ignoreCase = false): void
+    {
+        if (PHP_INT_SIZE == 8) {
+            // On 64-bit systems, compare integers directly
+            $expectedValue = (int) $expected;
+            $actualValue = (int) $actual;
+        } else {
+            // On 32-bit systems, compare integers as strings
+            $expectedValue = (string) $expected;
+            $actualValue = (string) $actual;
+        }
+
+        if ($expectedValue === $actualValue) {
+            return;
+        }
```

Comment:
Is this necessary given how comparisons for Int64 work? Would it be preferable to use `==` here, or might that cause problems on 32-bit platforms?

[bson-int64-compare-001.phpt](https://github.com/mongodb/mongo-php-driver/blob/master/tests/bson/bson-int64-compare-001.phpt) doesn't test comparisons against other types, but I think this is handled via the cast handlers (also added in mongodb/mongo-php-driver/commit/b1ff48a00c831cd1f18d6ab9b94ec2c61002162d for [PHPC-2214](https://jira.mongodb.org/browse/PHPC-2214)). Consider:

```
> 5 == new MongoDB\BSON\Int64(5)
= true

> 5.2 == new MongoDB\BSON\Int64(5)
= false

> '5.0' == new MongoDB\BSON\Int64(5)
= true

> 5.0 == new MongoDB\BSON\Int64(5)
= true

> '5.2' == new MongoDB\BSON\Int64(5)
= false
```

My concern for 32-bit platforms is rooted in the string casting. Consider:

```
> (string) new MongoDB\BSON\Int64(5) === (string) '5.0'
= false

> (int) new MongoDB\BSON\Int64(5) === (int) '5.0'
= true
```

`assertEquals()` should behave consistently on any platform.

This should also provide you with a bunch of new regression tests with fractional floats and float-like string. And if there's some way to test the 32-bit branch on 64-bit platforms, that might also help. As-is, I assume we have no actual test coverage of that code path.